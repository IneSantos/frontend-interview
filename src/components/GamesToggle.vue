
<template>
  <div class="gamestoggle">
    <button
        class="toggle"
        @click="toggle"
        :class="[showTicTacToe ? 'active' : '']"
        ref="tic-tac-toe"
    >Tic tac toe</button>
    <button
        class="toggle"
        @click="toggle"
        :class="[!showTicTacToe ? 'active' : '']"
        ref="4-in-a-row"
    >4-in-a-row</button>
</div>
</template>


<script>
import '@/styles/gamesToggle.scss';
export default {
  name: 'GamesToggle',
    data() {
        return {
            showTicTacToe: true
        };
    },
    methods: {
        toggle() {
            this.showTicTacToe = !this.showTicTacToe;
            this.$emit('toggle', this.showTicTacToe);
        },
    }
}
</script>