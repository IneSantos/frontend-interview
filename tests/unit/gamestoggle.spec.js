import { mount, shallowMount } from '@vue/test-utils'
import GamesToggle from '@/components/GamesToggle.vue'

describe('GamesToggle.vue', () => {
  it('renders default', () => {
    const wrapper = shallowMount(GamesToggle);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should should emmit toggle when toggle button is clicked', async () => {
    const wrapper = mount( GamesToggle );

    await wrapper.findComponent({ ref: "tic-tac-toe" }).trigger('click');
    expect(wrapper.emitted().toggle).toBeTruthy();
  })

  it('Should should call toggle when toggle button is clicked', async () => {
    const mockMethod = jest.spyOn(GamesToggle.methods, 'toggle');
    const wrapper = mount( GamesToggle );

    await wrapper.findComponent({ ref: "tic-tac-toe" }).trigger('click');
    expect(mockMethod).toHaveBeenCalled();
  })
})
