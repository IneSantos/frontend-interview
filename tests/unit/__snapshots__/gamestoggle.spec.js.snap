// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesToggle.vue renders default 1`] = `<div class="gamestoggle"><button class="toggle active">Tic tac toe</button> <button class="toggle">4-in-a-row</button></div>`;
